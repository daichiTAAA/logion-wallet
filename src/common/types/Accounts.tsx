import moment, { Moment } from 'moment';
import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';

export interface Token {
    readonly value: string;
    readonly expirationDateTime: Moment;
}

export interface Account {
    readonly name: string,
    readonly address: string,
    readonly isLegalOfficer: boolean,
    readonly token?: Token,
}

export default interface Accounts {
    readonly all: Account[],
    readonly current?: Account,
}

export class AccountTokens {

    constructor(initialState: Record<string, Token>) {
        this.store = { ...initialState };
    }

    private store: Record<string, Token>;

    get(address: string): Token | undefined {
        return this.store[address];
    }

    merge(tokens: AccountTokens): AccountTokens {
        const newStore = { ...this.store };
        for(const address of tokens.addresses) {
            newStore[address] = tokens.store[address];
        }
        return new AccountTokens(newStore);
    }

    get addresses(): string[] {
        return Object.keys(this.store);
    }

    cleanUp(now: Moment): AccountTokens {
        const newStore: Record<string, Token> = {};
        for(const address of this.addresses) {
            const token = this.get(address)!;
            if(token.expirationDateTime.isAfter(now)) {
                newStore[address] = token;
            }
        }
        return new AccountTokens(newStore);
    }

    equals(other: AccountTokens): boolean {
        if(this.length !== other.length) {
            return false;
        }
        for(const address of this.addresses) {
            const thisToken = this.get(address);
            const otherToken = other.get(address);
            if(thisToken!.value !== otherToken?.value
                || !thisToken!.expirationDateTime.isSame(otherToken.expirationDateTime)) {
                return false;
            }
        }
        return true;
    }

    get length(): number {
        return this.addresses.length;
    }

    isAuthenticated(now: Moment, address: string | undefined): boolean {
        if(address === undefined) {
            return false;
        }
        const token = this.get(address);
        if(token === undefined) {
            return false;
        } else {
            return token.expirationDateTime.isAfter(now);
        }
    }

    earliestExpiration(): Moment | undefined {
        let earliest: Moment | undefined;
        for(const address of this.addresses) {
            const expiration = this.store[address].expirationDateTime;
            if(earliest === undefined || earliest.isAfter(expiration)) {
                earliest = expiration;
            }
        }
        return earliest;
    }
}

export function buildAccounts(
    injectedAccounts: InjectedAccountWithMeta[],
    userAddress: string | undefined,
    tokens: AccountTokens,
    isLegalOfficer: (address: string) => boolean,
): Accounts {
    const selectedAddress = currentOrDefaultAddress(injectedAccounts, userAddress, tokens);

    let current: Account | undefined;
    const matchingAddress = injectedAccounts.find(injectedAccount => injectedAccount.address === selectedAddress);
    if(selectedAddress !== undefined && matchingAddress !== undefined) {
        current = {
            name: matchingAddress.meta.name!,
            address: selectedAddress!,
            isLegalOfficer: isLegalOfficer(selectedAddress),
            token: tokenOrUndefinedIfExpired(tokens.get(selectedAddress)),
        }
    } else {
        current = undefined;
    }

    return {
        current,
        all: injectedAccounts.map(injectedAccount => ({
            name: injectedAccount.meta.name!,
            address: injectedAccount.address,
            isLegalOfficer: isLegalOfficer(injectedAccount.address),
            token: tokenOrUndefinedIfExpired(tokens.get(injectedAccount.address)),
        }))
    }
}

function currentOrDefaultAddress(
    injectedAccounts: InjectedAccountWithMeta[],
    currentAddress: string | undefined,
    tokens: AccountTokens
): string | undefined {
    const loggedAddresses = tokens.addresses;
    if(currentAddress !== undefined) {
        return currentAddress;
    } else if(loggedAddresses.length > 0) {
        const defaultAccount = injectedAccounts.find(account => loggedAddresses.includes(account.address));
        return defaultAccount?.address;
    } else {
        return undefined;
    }
}

function tokenOrUndefinedIfExpired(token: Token | undefined): Token | undefined {
    if(token === undefined) {
        return undefined;
    }

    const expirationDateTime = token.expirationDateTime;
    if(expirationDateTime.isBefore(moment())) {
        return undefined;
    } else {
        return token;
    }
}
