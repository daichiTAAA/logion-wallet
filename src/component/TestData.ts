import Identity from './types/Identity';
import PostalAddress from './types/PostalAddress';
import { ColorTheme, rgbaToHex } from '../component/ColorTheme';

export const DEFAULT_IDENTITY: Identity = {
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

export const DEFAULT_ADDRESS: PostalAddress = {
    line1: "Place de le République Française, 10",
    line2: "boite 15",
    postalCode: "4000",
    city: "Liège",
    country: "Belgium",
};

export const COLOR_THEME: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: '#eff3fe',
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: '#d8e2fd',
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: 'eff3fe',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
        altBackground: '#eff3fe',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondary: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
    select: {
        background: '#ffffff',
        foreground: '#000000',
        menuBackgroundColor: '#ffffff',
        selectedOptionBackgroundColor: '#ffffff',
    },
    table: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
    tabs: {
        background: '#0c163d',
        foreground: '#ffffff',
        inactiveTab: {
            background: '#d8e2fd',
            foreground: '#ffffff',
        },
        borderColor: '#3b6cf4',
    },
    dialog: {
        background: '#eff3fe',
        foreground: '#000000',
        borderColor: '#e6007a',
    },
};
