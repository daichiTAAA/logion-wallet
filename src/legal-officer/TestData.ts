import moment from 'moment';
import { toIsoString } from '../logion-chain/datetime';
import { ProtectionRequest, DEFAULT_LEGAL_OFFICER } from './Types';

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        decisions: [
            {
                legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
                status: 'PENDING',
                rejectReason: null,
                decisionOn: null
            }
        ],
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        decisions: [
            {
                legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
                status: 'ACCEPTED',
                rejectReason: null,
                decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
            }
        ],
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z'))
    }
];
