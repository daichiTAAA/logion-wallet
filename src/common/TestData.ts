import { Coin, CoinBalance, SYMBOL } from '../logion-chain/Balances';
import { PrefixedNumber, ATTO } from '../logion-chain/numbers';

import Identity from './types/Identity';
import PostalAddress from './types/PostalAddress';
import { Transaction } from './types/ModelTypes';
import { ColorTheme, rgbaToHex } from './ColorTheme';

export const DEFAULT_LEGAL_OFFICER = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"; // Alice
export const ANOTHER_LEGAL_OFFICER = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"; // Bob
export const A_THIRD_LEGAL_OFFICER = "5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y"; // Charlie

export const DEFAULT_IDENTITY: Identity = {
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

export const DEFAULT_ADDRESS: PostalAddress = {
    line1: "Place de le République Française, 10",
    line2: "boite 15",
    postalCode: "4000",
    city: "Liège",
    country: "Belgium",
};

export const COLOR_THEME: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: '#eff3fe',
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: '#d8e2fd',
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: 'eff3fe',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
        altBackground: '#eff3fe',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    recoveryItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondary: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
    select: {
        background: '#ffffff',
        foreground: '#000000',
        menuBackgroundColor: '#ffffff',
        selectedOptionBackgroundColor: '#ffffff',
    },
    table: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
    tabs: {
        background: '#0c163d',
        foreground: '#ffffff',
        inactiveTab: {
            background: '#d8e2fd',
            foreground: '#ffffff',
        },
        borderColor: '#3b6cf4',
    },
    dialog: {
        background: '#eff3fe',
        foreground: '#000000',
        borderColor: '#e6007a',
    },
};

export const DEFAULT_COIN: Coin = {
    id: 'log',
    iconId: 'log',
    iconType: 'png',
    name: "Logion",
    symbol: SYMBOL
};

export const DEFAULT_COIN_BALANCE: CoinBalance = {
    coin: DEFAULT_COIN,
    balance: new PrefixedNumber("42", ATTO),
    level: 0.1,
};

export const DEFAULT_TRANSACTION: Transaction = {
    from: "from",
    to: "to",
    pallet: "pallet",
    method: "method",
    transferValue: "420",
    tip: "0",
    fee: "125000",
    reserved: "0",
    total: "125420",
    createdOn: "2021-07-28T12:30:00.000",
    type: "Sent",
}
