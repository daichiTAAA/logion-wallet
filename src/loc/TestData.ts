import { LegalOfficerCase } from "@logion/node-api/dist/Types";
import { UUID } from "@logion/node-api/dist/UUID";

import { LocRequest } from "../common/types/ModelTypes";

export function buildLocRequest(locId: UUID, loc: LegalOfficerCase): LocRequest {
    return {
        ownerAddress: loc.owner,
        requesterAddress: loc.requesterAddress,
        requesterIdentityLoc: loc.requesterLocId?.toString(),
        description: "Description",
        locType: loc.locType,
        createdOn: "",
        decisionOn: "",
        id: locId.toString(),
        status: loc.closed ? "CLOSED" : "OPEN",
        files: loc.files.map((locFile, index) => ({
            ...locFile,
            name: `File ${index}`,
            addedOn: "",
        })),
        metadata: loc.metadata.map((locFile, index) => ({
            ...locFile,
            name: `Data ${index}`,
            addedOn: "",
        })),
        links: loc.links.map((locFile, index) => ({
            ...locFile,
            name: `Link ${index}`,
            addedOn: "",
            target: locFile.id.toString(),
        }))
    };
}
