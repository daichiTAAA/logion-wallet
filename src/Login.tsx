import { useState, useCallback } from 'react';
import { useHistory, useLocation } from 'react-router-dom';

import Container from 'react-bootstrap/Container';

import { useCommonContext } from './common/CommonContext';
import { useLogionChain } from './logion-chain';
import Button from './common/Button';
import Checkbox from './common/Checkbox';
import { authenticate } from './common/Authentication';

import AbsoluteLogo from './AbsoluteLogo';

import './Login.css';

export const LOGIN_PATH = "/login";

export interface LocationState {
    referrer?: string;
    selectedAddresses?: string[];
}

export default function Login() {
    const location = useLocation<LocationState>();
    const history = useHistory();
    const { connectedNodeMetadata } = useLogionChain();
    const { accounts, setTokens, axiosFactory } = useCommonContext();
    const [ selectedAddresses, setSelectedAddresses ] = useState<string[]>(location.state && location.state.selectedAddresses ? location.state.selectedAddresses : []);

    const startLogin = useCallback(async () => {
        const tokens = await authenticate(axiosFactory!(), selectedAddresses);
        setTokens(tokens);

        if(location.state && location.state.referrer) {
            history.push(location.state.referrer);
        } else {
            history.push("/");
        }
    }, [ axiosFactory, selectedAddresses, setTokens, history, location ]);

    if(accounts === null || connectedNodeMetadata === null || axiosFactory === undefined) {
        return null;
    }

    function selectAddress(address: string, selected: boolean) {
        let newSet;
        if(selected) {
            newSet = [ ...selectedAddresses ];
            newSet.push(address);
        } else {
            newSet = selectedAddresses.filter(item => item !== address);
        }
        setSelectedAddresses(newSet);
    }

    return (
        <div className="Login">
            <AbsoluteLogo />
            <Container className="column">
                <h1>
                    Please authenticate yourself by selecting<br />
                    one or several accounts below
                </h1>

                <div className="chooser">
                    <h2>The following accounts have been detected:</h2>

                    <div className="accounts-container">
                        {
                            accounts.all
                            .filter(address => address.token === undefined)
                            .map(address => (
                                <div className="account" key={ address.address }>
                                    <Checkbox
                                        checked={ selectedAddresses.includes(address.address) }
                                        setChecked={ selected => selectAddress(address.address, selected) }
                                    />
                                    <div
                                        className="icon"
                                    >
                                        { address.name.substring(0, 1).toUpperCase() }
                                    </div>
                                    <div
                                        className="name-address"
                                    >
                                        <div className="name">{ address.name }</div>
                                        <div className="address">{ address.address }</div>
                                    </div>
                                </div>
                            ))
                        }
                    </div>

                    <Button
                        disabled={ selectedAddresses.length === 0 }
                        onClick={ startLogin }
                    >
                        Log in
                    </Button>
                </div>

                <div className="node-info">
                    <p className="name">You are currently connected to the following logion node: "{connectedNodeMetadata.name}"</p>
                    <p className="peer-id">
                        <img src={process.env.PUBLIC_URL + "/assets/node.svg"} alt="node icon" />
                        Peer ID: {connectedNodeMetadata.peerId}
                    </p>
                </div>
            </Container>
            <div className="left-character">
                <img src={process.env.PUBLIC_URL + "/assets/login-left-character.svg"} alt="men pointing at box" />
            </div>
            <div className="right-character">
                <img src={process.env.PUBLIC_URL + "/assets/login-right-character.svg"} alt="women holding a loc" />
            </div>
        </div>
    );
}
