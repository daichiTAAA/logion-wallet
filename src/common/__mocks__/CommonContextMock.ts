import { TEST_WALLET_USER } from '../../wallet-user/TestData';
import { COLOR_THEME } from '../TestData';
import Addresses, { AccountAddress } from '../types/Addresses';

export let selectAddress = jest.fn();

export let currentAddress = TEST_WALLET_USER;

export function setCurrentAddress(address: string) {
    currentAddress = address;
}

const DEFAULT_ACCOUNT: AccountAddress = {
    name: "name",
    address: TEST_WALLET_USER,
    isLegalOfficer: false,
}

export let addresses: Addresses = {
    currentAddress: DEFAULT_ACCOUNT,
    addresses: [ DEFAULT_ACCOUNT ]
}

export let balances: any = null;

export let transactions: any = null;

export function useCommonContext() {
    return {
        currentAddress,
        selectAddress,
        addresses,
        balances,
        transactions,
        colorTheme: COLOR_THEME,
    };
}

export function setAddresses(value: Addresses) {
    addresses = value;
}

export function setBalances(value: any) {
    balances = value;
}

export function setTransactions(value: any) {
    transactions = value;
}
