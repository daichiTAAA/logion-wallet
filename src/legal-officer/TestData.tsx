import moment from 'moment';
import { toIsoString } from '@logion/node-api/dist/datetime';
import { ProtectionRequest } from '@logion/client/dist/RecoveryClient';

import { DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER } from "../common/TestData";

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
            rejectReason: null,
            decisionOn: null,
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: false,
        addressToRecover: null,
        status: "PENDING"
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
                rejectReason: null,
                decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
                locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: false,
        addressToRecover: null,
        status: "ACTIVATED"
    }
];

export const RECOVERY_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER,
        decision: {
            rejectReason: null,
            decisionOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
            locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(moment('2021-06-10T11:40:00.000Z')),
        isRecovery: true,
        addressToRecover: "an-address",
        status: "PENDING"
    }
];
