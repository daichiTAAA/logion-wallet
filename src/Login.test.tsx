import moment from 'moment';
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Accounts, { Account } from './common/types/Accounts';
import { shallowRender } from './tests';
import { setAddresses, authenticate } from './common/__mocks__/CommonContextMock';
import { TEST_WALLET_USER } from './wallet-user/TestData';
import { DEFAULT_LEGAL_OFFICER } from "./common/TestData";
import Login from './Login';

jest.mock('./logion-chain');
jest.mock('./common/CommonContext');
jest.mock('./common/Authentication');

const AUTHENTICATED_ADDRESS: Account = {
    name: "name authenticated",
    address: TEST_WALLET_USER,
    isLegalOfficer: false,
    token: {
        value: "token",
        expirationDateTime: moment('2021-09-10T10:53:00.000'),
    },
};

const UNAUTHENTICATED_ADDRESS: Account = {
    name: "name unauthenticated",
    address: DEFAULT_LEGAL_OFFICER,
    isLegalOfficer: true,
};

const ADDRESSES: Accounts = {
    current: AUTHENTICATED_ADDRESS,
    all: [ UNAUTHENTICATED_ADDRESS ]
};

describe("Login", () => {

    it("renders", () => {
        setAddresses(ADDRESSES);
        const result = shallowRender(
            <Login />
        );
        expect(result).toMatchSnapshot();
    });

    it("enables log-in given unautenticated yet user", async () => {
        setAddresses(ADDRESSES);
        render(
            <Login />
        );

        await userEvent.click(screen.getByRole('checkbox'));
        await userEvent.click(screen.getByRole('button', {name: "Log in"}));

        await waitFor(() => expect(authenticate).toBeCalledWith([ DEFAULT_LEGAL_OFFICER ]));
    });
});
