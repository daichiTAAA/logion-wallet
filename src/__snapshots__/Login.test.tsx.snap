// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders 1`] = `
<div
  className="Login"
>
  <AbsoluteLogo />
  <Container
    className="column"
    fluid={false}
  >
    <h1>
      Please authenticate yourself by selecting
      <br />
      one or several accounts below
    </h1>
    <div
      className="chooser"
    >
      <h2>
        The following accounts have been detected:
      </h2>
      <div
        className="accounts-container"
      >
        <div
          className="account"
        >
          <Checkbox
            checked={false}
            setChecked={[Function]}
          />
          <div
            className="icon"
          >
            N
          </div>
          <div
            className="name-address"
          >
            <div
              className="name"
            >
              name unauthenticated
            </div>
            <div
              className="address"
            >
              5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
            </div>
          </div>
        </div>
      </div>
      <Button
        disabled={true}
        onClick={[Function]}
      >
        Log in
      </Button>
    </div>
    <div
      className="node-info"
    >
      <p
        className="name"
      >
        You are currently connected to the following logion node: "
        "
      </p>
      <p
        className="peer-id"
      >
        <img
          alt="node icon"
          src="/assets/node.svg"
        />
        Peer ID: 
      </p>
    </div>
  </Container>
  <div
    className="left-character"
  >
    <img
      alt="men pointing at box"
      src="/assets/login-left-character.svg"
    />
  </div>
  <div
    className="right-character"
  >
    <img
      alt="women holding a loc"
      src="/assets/login-right-character.svg"
    />
  </div>
  <Dialog
    actions={
      Array [
        Object {
          "buttonText": "Ignore",
          "buttonVariant": "secondary",
          "callback": [Function],
          "id": "ignore",
        },
      ]
    }
    show={false}
    size="lg"
  >
    <h2>
      Upgrade to the latest app version
    </h2>
    <p>
      A new version of logion is available (
      0
      ) but your are currently using version 
      0
      .
    </p>
    <p>
      In order to get the latest version, please request a "hard-refresh" (i.e. clear your cache and refresh) to your browser now:
    </p>
    <p>
      <strong>
        PC users: Ctrl-F5
      </strong>
    </p>
    <p>
      <strong>
        Mac users: Cmd-Shift-R
      </strong>
    </p>
    <p>
      Staying up to date is important in order to benefit from functional and security fixes.
    </p>
    <p>
      What changed since last version:
    </p>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </Dialog>
</div>
`;
